src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -57,110 +38,110 @@ public class ParseAndCompareBenchmarkResults {
-     * @param benchmarkResultNew      is the benchmark result of ion-java from the new commit.
-     * @return a map which match the thresholds score with the aspect name it represents.
-     * @throws Exception if errors occurs when calling method parseScore.
+     * This method composes the results of the performance regression detection.
+     * @param result represents the metrics which have regressions and their regression values.
+     * @return the composed regression detection summary.
      */
-    public static Map<String, BigDecimal> getThresholdMap(String benchmarkResultPrevious, String benchmarkResultNew) throws Exception {
-        Map<String, BigDecimal> thresholdMap = new HashMap<>();
-        for (String keyWord : BENCHMARK_SCORE_KEYWORDS) {
-            IonList rawDataPrevious = (IonList) parseScore(benchmarkResultPrevious, keyWord).get(RAW_DATA);
-            IonList rawDataNew = (IonList) parseScore(benchmarkResultNew, keyWord).get(RAW_DATA);
-            BigDecimal thresholdPrevious = getThresholdScore((IonList) rawDataPrevious.get(0));
-            BigDecimal thresholdNew = getThresholdScore((IonList) rawDataNew.get(0));
-            if (thresholdPrevious.compareTo(thresholdNew) < 0) {
-                thresholdMap.put(keyWord, thresholdPrevious);
-            } else {
-                thresholdMap.put(keyWord, thresholdNew);
-            }
+    private static String comparisonSummary(Map<String, Double> result) {
+        StringBuilder summary = new StringBuilder();
+        for (Map.Entry<String, Double> comparisonResult : result.entrySet()) {
+            summary.append(String.format("There is %.2f%% regression on %s.\n", comparisonResult.getValue() * 100, comparisonResult.getKey()));
         }
-        return thresholdMap;
+        return summary.toString();
     }
 
     /**
-     * Get threshold score by applying (minScore - maxScore)/maxScore to a list of raw data in benchmark result.
-     *
-     * @param rawDataList is an Ion List contains performance scores from multiple iterations of benchmark process.
-     * @return calculated threshold score.
+     * This method uses two-sample T-test method to detect performance regression.
+     * @param before represents the benchmark results from benchmarking the previous ion-java.
+     * @param after represents the benchmark results from benchmarking the ion-java with the new changes.
+     * @return regression detection result. If the regression is detected, then the regression value will be returned, else 0 will be returned.
      */
-    private static BigDecimal getThresholdScore(IonList rawDataList) {
-        List<BigDecimal> rawData = new ArrayList<>();
-        for (int i = 0; i < rawDataList.size(); i++) {
-            IonDecimal score = (IonDecimal) rawDataList.get(i);
-            rawData.add(score.bigDecimalValue());
+    public static double detectRegression(double[] before, double[] after) {
```

Comment:
Technically, this is detecting whether there is a statistically significant change. It looks like it will return a non-zero value both for a significant regression and a significant improvement.

Reply:
> Technically, this is detecting whether there is a statistically significant change. It looks like it will return a non-zero value both for a significant regression and a significant improvement.

Yes, and only significant regression value will be added to the `comparisonResults` map which is used for deciding whether `ion-java-regression-detection` workflow should fail or not. Here is the [comment](https://github.com/amazon-ion/ion-java-benchmark-cli/pull/58/files/2b02d3a1cd79441eff09db1aa3a2ba4c23cc460e#r1355716936) about the conditional check